Classes/EventStore/AbstractEventSourcedRepository.php

```diff
@@ -84,28 +78,25 @@ public function findByIdentifier($identifier)
     }
 
     /**
-     * @param  AggregateRootInterface $aggregate
+     * @param AggregateRootInterface $aggregate
+     * @param int $expectedVersion
      * @return void
      */
-    public function save(AggregateRootInterface $aggregate)
+    final public function save(AggregateRootInterface $aggregate, int $expectedVersion = null)
```

Comment:
Hm, isn't the expected version already implicitly defined by the version of the first event in the `uncommittedEvents`? I mean it always expects to be at the version **before** the first uncommitted Event and anything else is a ConcurrencyException.